### ISSUE TITLE
Support owc/1.0/

### ISSUE THREAD

**Issue by torse** [highsource/ogc-schemas#47]
http://schemas.opengis.net/owc/1.0/

**Comment by highsource** [comment]
Just to make it sure, you really need this:

http://schemas.opengis.net/owc/1.0/

And not this:

http://schemas.opengis.net/context/

Please confirm. OWC is "OGC Web Services Context Atom Standard", in RelaxNG.

**Comment by torse** [comment]
Yes, your first link is correct. We would need JSON bindings for OWC 1.0.0 (OWS Context or OGC Web Services Context Document) which is the successor to WMC 1.1.0/1.0.0. For a short description see: http://www.ogcnetwork.net/context

**Comment by highsource** [comment]
Ok, thanks for the confirmation.

This is RelaxNG Compact, not an XML schema, I don't have aneough experience here - but I'll see what I can do.

**Comment by torse** [comment]
It's appreciated, thanks! But don't spend too much time on it, as it is not critical for us right now. Also, I just found that there are also (Geo)JSON binding rules defined as XSLT in a OGC PER, see: http://www.opengeospatial.org/standards/owc.

**Comment by highsource** [comment]
I think Jsonix will be able to do this pretty well. I'll see what I can do.

**Comment by highsource** [comment]
If you happen to have examples of XML/JSON pairs that would be very helpful. There are XML examples [here](http://schemas.opengis.net/owc/1.0/examples/) but would be good to have JSON as well.
Maybe with this XSLT...

**Comment by highsource** [comment]
@torse As feared, everything's not easy:

* highsource/maven-jaxb2-plugin#42

which is due to

* https://java.net/jira/browse/JAXB-1065

**Comment by highsource** [comment]
@torse 

Even worse than that:

* https://java.net/jira/browse/JAXB-1066
* https://java.net/jira/browse/JAXB-1067

So RELAXNG compilation does not work (at all), unfortunately.

The option here is to create and compile XML Schemas.

**Comment by torse** [comment]
Ok, tried to convert this RELAXNG with trang (https://code.google.com/p/jing-trang/) to XSD. Got the following warnings. Not sure if that XSDs are safe to map?

```
$ java -jar trang-20091111/trang.jar owc/1.0.0/owc.rnc owc.xsd
/local/schemas/owc/1.0.0/owc.rnc:53:60: warning: choice between attributes and children cannot be represented; approximating
/local/schemas/owc/1.0.0/georss.rnc:39:16: warning: cannot represent required wildcard attribute; approximating
/local/schemas/owc/1.0.0/owc.rnc:87:12: warning: cannot represent required wildcard attribute; approximating
/local/schemas/owc/1.0.0/owc.rnc:88:9: warning: choice between attributes and children cannot be represented; approximating
/local/schemas/owc/1.0.0/atom_feed.rnc:26:44: warning: choice between attributes and children cannot be represented; approximating
/local/schemas/owc/1.0.0/atom_feed.rnc:173:22: warning: cannot represent required wildcard attribute; approximating
/local/schemas/owc/1.0.0/atom_feed.rnc:177:55: warning: choice between attributes and children cannot be represented; approximating
/local/schemas/owc/1.0.0/atom_feed.rnc:179:71: warning: choice between attributes and children cannot be represented; approximating
/local/schemas/owc/1.0.0/atom_feed.rnc:182:59: warning: choice between attributes and children cannot be represented; approximating
/local/schemas/owc/1.0.0/atom_feed.rnc:184:61: warning: choice between attributes and children cannot be represented; approximating
```

**Comment by torse** [comment]
Not sure if this helps, but here are some related links..

XSL to transform the OWC XML to JSON:
https://github.com/opengeospatial/owscontext/tree/master/json/dev/trax/xslt

JSON versions of the official OWC examples transform with the above XSL:
https://github.com/opengeospatial/owscontext/tree/master/json/dev/trax/json

**Comment by highsource** [comment]
This would be good help, could you please send me the XSDs? I'll check bout the safety.

**Comment by torse** [comment]
Please find the trang generated XSD of OWC attached..

**Comment by highsource** [comment]
@torse Would you please mail to valikov-at-gmx-net, attachment did not went through GitHub.

**Comment by highsource** [comment]
Ok, just converted.
